python/ray/tests/test_state_api.py

```diff
@@ -2068,6 +2068,96 @@ def ready(self):
     assert "actor_id" in result.output
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason="Lambda test functions could not be pickled on Windows",
+)
+@pytest.mark.parametrize(
+    "api_func",
+    [
+        # NOTE(rickyyx): arbitrary list of APIs, not exhaustive.
+        list_objects,
+        list_tasks,
+        list_actors,
+        list_nodes,
+        list_placement_groups,
+    ],
+)
+def test_state_api_server_enforce_concurrent_http_requests(api_func):
+    import time
+    import multiprocessing as mp
+    import queue
+    import os
+
+    # Set environment
+    os.environ["RAY_STATE_SERVER_MAX_HTTP_REQUEST"] = "1"
+
+    ray.shutdown()
+    ray.init()
+
+    # Set up scripts
+    @ray.remote
+    def f():
+        import time
+
+        time.sleep(30)
+
+    @ray.remote
+    class Actor:
+        pass
+
+    task = f.remote()  # noqa
+    actor = Actor.remote()  # noqa
+    actor_runtime_env = Actor.options(  # noqa
+        runtime_env={"pip": ["requests"]}
+    ).remote()
+    pg = ray.util.placement_group(bundles=[{"CPU": 1}])  # noqa
+
+    _objs = [ray.put(x) for x in range(10)]  # noqa
+
+    # Wait for results to be populate on the cluster
```

Comment:
what does it mean by result? Can we use wait_for_condition instead of sleep?

Reply:
I guess this is to ensure the state api query will not race with those operations above.